### PR TITLE
so that we create less turds, add a `datastore` field to the projects table and add logic so that db containers arent created when the field is nil

### PR DESCRIPTION
**Pull request by alexebird**
DRY up build templating
[Tracker #102097496](https://www.pivotaltracker.com/story/show/102097496)

### CODE REVIEW

File: lib/integrity/build.rb [hunk 1]
```diff
@@ -35,8 +35,8 @@ class Build
     property :state,                 String
     property :fleetctl_status,       String
     property :machine_ip,            String
-    property :service_template,      String, default: 'spec'
-    property :public_port,           String
+    #property :service_template,      String, default: 'spec'
```
Comment (RoryO) [inline lib/integrity/build.rb]: leave these in?

File: lib/integrity/build/status.rb [hunk 2]
```diff
@@ -123,13 +123,12 @@ def bootstrap_class
       end
 
       def update_fleetctl_status(status)
-        if app_status = status.detect{|s| s[:type] == 'app' }
-          update(fleetctl_status: app_status[:active_state], machine_ip: app_status[:machine_ip])
-        end
+        #if app_status = status.detect{|s| s[:type] == 'app' }
```
Comment (RoryO) [inline lib/integrity/build/status.rb]: also leave these in?

File: lib/pancakes/fleet/templating/honesty_builds.rb [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+        #####   #####      #    #    #    ##     #####  ######
+        #    #  #    #     #    #    #   #  #      #    #
+        #    #  #    #     #    #    #  #    #     #    #####
+        #####   #####      #    #    #  ######     #    #
+        #       #   #      #     #  #   #    #     #    #
+        #       #    #     #      ##    #    #     #    ######
+
+
+        # @api private
```
Comment (RoryO) [inline lib/pancakes/fleet/templating/honesty_builds.rb]: I don't get it... are these supposed to be private methods? Why not use private def fff()?

Comment (alexebird) [inline lib/pancakes/fleet/templating/honesty_builds.rb]: @RoryO It's fun using `banner`. Good point though. But why is activesupport::concern required to use private?

**Comment by cobbr2** [comment]
This seems to be doing a whole lot more than just adding a property to a class and adding an if statement, which is what I would expect from the summary. Is this PR based on some other story in which we remove the old "server" containers, etc.,?  Overall, I don't see a big problem with the code (well, the fact that there are no tests makes me sad), but what is this patch *actually trying to do?*

**Comment by alexebird** [comment]
@cobbr2 Yes, the patch is pretty large relative to the story, and it's true that some of it is unrelated to the story, but not everything. Adding the `datastore` field caused the old way of determining the datastore to become dead, which led me to more dead code, which lead me to more, until I realized that should probably be a separate effort. But I didn't want to throw out the progress I'd already made.

The other part of the explanation is that the Fleet templating module was based on the assumption that builds *always* have datastore container, which is not true and is the purpose of this story. It's been clear that the build templating code has needed some love for a while now, and modifying it to handle no datastore container led to some refactoring, which led to more, which led to addressing these long-standing pain points in the module.

As for the tests, I agree, but since no-tests has been part of the Honesty culture since day 1, I just kept in line with that. But in the future I think any new Honesty modules/classes should have tests.

**Pull request merged**